src/Bridge.sol

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import "./TokenBase.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "forge-std/console.sol";
+
+contract Bridge is Ownable {
+    mapping(address => address) public tokenPairs;
+
+    fallback() external payable {}
+    receive() external payable {}
+
+    constructor(address _initialOwner) Ownable(_initialOwner) {}
+
+    event TokenLocked(address _from, address _erc20token, uint256 _ammount);
+    event TokenReleased(address _to, address _erc20token, uint256 _ammount);
+    event TokenBurned(address _from, address _erc20token, uint256 _ammount);
+    event TokenMinted(address _to, address _erc20token, uint256 _ammount);
+
+    function lock(
+        address _from,
+        address _erc20token,
+        uint256 _ammount
+    ) external {
+        require(
+            TokenBase(_erc20token).transferFrom(_from, address(this), _ammount),
+            "Locking failed"
+        );
+
+        emit TokenLocked(_from, _erc20token, _ammount);
+    }
+
+    function release(
+        address _to,
+        address _erc20token,
+        uint256 _ammount
+    ) external onlyOwner {
+        require(
+            TokenBase(_erc20token).transfer(_to, _ammount),
+            "Release failed"
+        );
+
+        emit TokenReleased(_to, _erc20token, _ammount);
+    }
+
+    function burn(
+        address _from,
+        address _erc20token,
+        uint256 _ammount
+    ) external {
+        TokenBase(_erc20token).burnFrom(_from, _ammount);
+
+        emit TokenBurned(_from, _erc20token, _ammount);
```

Comment:
Since you have a mapping of wrapped ERC-20 -> original ERC-20, you can emit the native token address in the event so that you know which token to transfer upon release.